src/Data/List/Relation/Split.agda

```diff
@@ -0,0 +1,109 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- General notion of splitting a list in two in an order-preserving manner
+------------------------------------------------------------------------
+
+module Data.List.Relation.Split where
+
+open import Level
+open import Data.List.Base as List using (List; []; _∷_)
+open import Data.List.Relation.Pointwise using (Pointwise; []; _∷_)
+open import Data.List.Relation.Permutation.Inductive as Perm using (_↭_)
+open import Data.List.Relation.Permutation.Inductive.Properties using (shift)
+open import Data.Product as Prod using (∃; ∃₂; _×_; uncurry; _,_; -,_; proj₂)
+open import Data.Sum as Sum using (_⊎_; inj₁; inj₂)
+open import Function
+open import Relation.Binary
+open import Relation.Binary.PropositionalEquality as P using (_≡_)
+
+module _ {a b c l r} {A : Set a} {B : Set b} {C : Set c}
+         (L : REL A B l) (R : REL A C r) where
+
+  data Split : List A → REL (List B) (List C) (l ⊔ r) where
+    []   : Split [] [] []
+    _ˡ∷_ : ∀ {a b as l r} → L a b → Split as l r → Split (a ∷ as) (b ∷ l) r
+    _ʳ∷_ : ∀ {a b as l r} → R a b → Split as l r → Split (a ∷ as) l (b ∷ r)
```

Comment:
Is there a reason that the superscripts are on the left? We usually have them on the right...

Reply:
I don't know why I did that. In [typing with leftovers](https://gallais.github.io/typing-with-leftovers/linear.Context.Mix.html), I do have them on the right.
I'll push a patch.